src/idaes_compatibility/robustness/unit_models/test_pressure_changer.py

```diff
@@ -0,0 +1,137 @@
+
+
+currdir = this_file_dir()
+fname = os.path.join(currdir, "isothermal_pressure_changer.json")
+
+
+@cb.register_convergence_class("IsothermalPressureChanger")
+class IsothermalPressureChangerConvergenceEvaluation(cb.ConvergenceEvaluation):
+    def get_specification(self):
+        """
+        Returns the convergence evaluation specification for the
+        isothermal PressureChanger unit model
+
+        Returns
+        -------
+           ConvergenceEvaluationSpecification
+        """
+        s = cb.ConvergenceEvaluationSpecification()
+
+        s.add_sampled_input(
+            name="Inlet_Flowrate",
+            pyomo_path="fs.pc.control_volume.properties_in[0].flow_mass",
+            lower=1,
+            upper=1e6,
+            distribution="uniform",
+        )
+
+        s.add_sampled_input(
+            name="Inlet_Pressure",
+            pyomo_path="fs.pc.control_volume.properties_in[0].pressure",
+            lower=10,
+            upper=2e8,
+            distribution="uniform",
+        )
+
+        s.add_sampled_input(
+            name="Inlet_Enthalpy",
+            pyomo_path="fs.pc.control_volume.properties_in[0].enth_mass",
+            lower=2e4,
+            upper=4.4e6,
+            distribution="uniform",
+        )
+
+        s.add_sampled_input(
+            name="Pressure_Change",
+            pyomo_path="fs.pc.deltaP[0]",
+            lower=-5e5,
+            upper=5e5,
+            distribution="uniform",
+        )
+
+        # TODO: Add deltaP as an input?
+        return s
+
+    def get_initialized_model(self):
+        """
+        Returns an initialized model for the PressureChanger unit model
+        convergence evaluation
+
+        Returns
+        -------
+           Pyomo model : returns a pyomo model of the PressureChanger unit
+        """
+        m = pe.ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+        m.fs.props = pp.Iapws95ParameterBlock(amount_basis=pp.AmountBasis.MASS)
+
+        m.fs.pc = PressureChanger(
+            property_package=m.fs.props,
+            thermodynamic_assumption=ThermodynamicAssumption.isothermal,
+        )
+
+        m.fs.pc.deltaP.fix(-1e3)
+        m.fs.pc.inlet[:].flow_mass.fix(27.5e3)
+        m.fs.pc.inlet[:].enth_mass.fix(4000)
+        m.fs.pc.inlet[:].pressure.fix(2e6)
+
+        init_state = {"flow_mass": 27.5e3, "pressure": 2e6, "enth_mass": 4000}
+
+        m.fs.pc.initialize(state_args=init_state)
+
+        # Create a solver for initialization
+        opt = self.get_solver()
+        opt.solve(m)
+
+        # return the initialized model
+        return m
+
+
+def test_isothermal_pressure_changer_robustness():
+    ceval = IsothermalPressureChangerConvergenceEvaluation()
+
+    # Uncomment to create new baseline file
+    # ceval.write_baseline_file(fname, 100)
+
+    solves, iters, restoration, regularization = ceval.compare_to_baseline(fname)
+
+    assert solves == []
+    assert iters == []
```

Comment:
Are you saying here that the number of iterations doesn't change?  That may be too much to ask, since it may vary depending on the platform and solver build. Maybe this model is simple enough that it's okay, but generally it seems a bit delicate.

Reply:
Yes, that is what it is saying and there is the concern that it will be fragile, but I am not sure what else we can do.

That said, I do not think we need to test this on all platforms, so we can at least avoid OS related issues. Beyond that, I would like to hope that the number of iterations is fairly stable.

Reply:
I see that we probably want to check that something doesn't happen where the model goes from taking like 5 iterations to solve to like 100, but I have a feeling this is going to be trouble.  If we are just running this test on one particular system, it probably should be fine.  I guess it depends on what we want to do with the test and who may end up running it.